app/services/umrdr/date_range_service.rb

```diff
@@ -0,0 +1,73 @@
+module Umrdr
+  class DateRangeService  
+
+    attr_reader :params
+    
+    def initialize(params)
+      @params = params
+    end  
+
+    def transform
+
+      if (params[:date_coverage_1_year].length>0)
+        yyyy1= params[:date_coverage_1_year].to_i 
+      else
+        yyyy2 = 0 #error 
+      end  
+      if (!params[:date_coverage_1_month].eql?("--"))
```

Comment:
negated `if` like this is usually written as `unless` in ruby.
The parenthesis here are superflous.
Use `==` instead of `eql?` See: http://ruby-doc.org/core-2.3.1/Object.html#method-i-eql-3F